File: Byte&Buy.AngularSPA/src/app/core/dto/offers/sale/user-sale-offer-response.ts
import { Guid } from "guid-typescript";
import { MoneyDto } from "../../common/money-dto";
import { ImageResponse } from "../../image/image-response";
import { OfferAddressResponse } from "../common/offer-address-response";

export interface UserSaleOfferResponse {
    id: Guid;
    categoryId: Guid;
    conditionId: Guid;
    name: string;
    description: string;
    quantityAvailable: number;
    offerAddress: OfferAddressResponse;
    pricePerItem: MoneyDto;
    images: ImageResponse[];
    parcelLockerDeliveries: Guid[] | null;
    otherDeliveriesIds: Guid[];
}

File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Offer/SaleOffer/UserSaleOfferResponse.cs
using ByteBuy.Core.DTO.Public.Image;
using ByteBuy.Core.DTO.Public.Money;
using ByteBuy.Core.DTO.Public.Offer.Common;

namespace ByteBuy.Core.DTO.Public.Offer.SaleOffer;

public record UserSaleOfferResponse(
   Guid Id,
   Guid CategoryId,
   Guid ConditionId,
   string Name,
   string Description,
   int QuantityAvailable,
   OfferAddressResponse OfferAddress,
   MoneyDto PricePerItem,
   IReadOnlyCollection<ImageResponse> Images,
   IReadOnlyCollection<Guid>? ParcelLockerDeliveries,
   IReadOnlyCollection<Guid> OtherDeliveriesIds);